src/metrics/datetime.ts

```diff
@@ -0,0 +1,227 @@
+    if (!isObject(v) || Object.keys(v).length !== 3) {
+      return false;
+    }
+
+    const timeUnitVerification = "timeUnit" in v && isString(v.timeUnit) && Object.values(TimeUnit).includes(v.timeUnit as TimeUnit);
+    const timezoneVerification = "timezone" in v && isNumber(v.timezone);
+    const dateVerification = "date" in v && isString(v.date) && v.date.length === 24 && !isNaN(Date.parse(v.date));
+    if (!timeUnitVerification || !timezoneVerification || !dateVerification) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Gets this metrics value in its payload representation.
+   *
+   * For this metric, the payload is the timezone aware ISO date string truncated to the time unit
+   * given at the time of recording.
+   *
+   * @returns The metric value.
+   */
+  payload(): string {
+    const timezone = formatTimezoneOffset(this.timezone);
+
+    const year = this.date.getFullYear().toString().padStart(2, "0");
+    const month = this.date.getMonth().toString().padStart(2, "0");
+    const day = this.date.getDate().toString().padStart(2, "0");
+    if (this.timeUnit === TimeUnit.Day) {
+      return `${year}-${month}-${day}${timezone}`;
+    }
+
+    const hours = this.date.getHours().toString().padStart(2, "0");
+    if (this.timeUnit === TimeUnit.Hour) {
+      return `${year}-${month}-${day}T${hours}${timezone}`;
+    }
+
+    const minutes = this.date.getMinutes().toString().padStart(2, "0");
+    if (this.timeUnit === TimeUnit.Minute) {
+      return `${year}-${month}-${day}T${hours}:${minutes}${timezone}`;
+    }
+
+    const seconds = this.date.getSeconds().toString().padStart(2, "0");
+    if (this.timeUnit === TimeUnit.Second) {
+      return `${year}-${month}-${day}T${hours}:${minutes}:${seconds}${timezone}`;
+    }
+
+    const milliseconds = this.date.getMilliseconds().toString().padStart(3, "0");
+    if (this.timeUnit === TimeUnit.Millisecond) {
+      return `${year}-${month}-${day}T${hours}:${minutes}:${seconds}.${milliseconds}${timezone}`;
+    }
+
+    if (this.timeUnit === TimeUnit.Microsecond) {
+      return `${year}-${month}-${day}T${hours}:${minutes}:${seconds}.${milliseconds}000${timezone}`;
+    }
+
+    return `${year}-${month}-${day}T${hours}:${minutes}:${seconds}.${milliseconds}000000${timezone}`;
+  }
+}
+
+/**
+ * A datetime metric.
+ *
+ * Used to record an absolute date and time, such as the time the user first ran
+ * the application.
+ */
+class DatetimeMetricType extends MetricType {
+  private timeUnit: TimeUnit;
+
+  constructor(meta: CommonMetricData, timeUnit: string) {
+    super("datetime", meta);
+    this.timeUnit = timeUnit as TimeUnit;
+  }
+
+  /**
+   * Sets to the specified datetime value.
+   *
+   * @param value the value to set.
+   */
+  async set(value?: Date): Promise<void> {
+    if (!this.shouldRecord()) {
+      return;
+    }
+
+    if (!value) {
+      value = new Date();
+    }
+
+    const metric =  new DatetimeMetric({
```

Comment:
```suggestion
    const metric = new DatetimeMetric({
```

Does eslint support a rule to detect "double spacing"?